turbopack/crates/turbopack-core/src/chunk/chunking/production.rs

```diff
@@ -53,17 +53,22 @@ pub async fn make_production_chunks(
             grouped_chunk_items.entry(key).or_default().push(chunk_item);
         }
 
-        let &ChunkingConfig { min_chunk_size, .. } = chunking_config;
+        let &ChunkingConfig {
+            min_chunk_size,
+            max_chunk_count_per_group,
+            max_merge_chunk_size,
+            ..
+        } = chunking_config;
 
-        if min_chunk_size == 0 {
+        if min_chunk_size == 0 && max_chunk_count_per_group == 0 {
             span.record("chunks", grouped_chunk_items.len());
             for chunk_items in grouped_chunk_items.into_values() {
                 make_chunk(chunk_items, &mut String::new(), &mut split_context).await?;
             }
         } else {
             let mut heap = BinaryHeap::new();
```

Comment:
(Unrelated to your changes though)
```suggestion
            let mut heap = BinaryHeap::with_capacity(grouped_chunk_items.len());
```